### PR TITLE
features/CIV-5563-Defendant's response- Full Dispute

### PR DESCRIPTION
**Pull request by laperla11**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-5563

### Change description ###
Develop the "View the defendants response" screen in the scenario where the Defendant has fully disputed the claim

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/common/models/claim.ts [hunk 1]
```diff
@@ -40,6 +40,13 @@ import {Interest} from '../form/models/interest/interest';
 import {RejectAllOfClaimType} from '../../common/form/models/rejectAllOfClaimType';
 import {ClaimDetails} from '../../common/form/models/claim/details/claimDetails';
 import {ClaimantResponse} from './claimantResponse';
+import {SelfEmployedAs} from '../models/selfEmployedAs';
```
Comment (laperla11) [inline src/main/common/models/claim.ts:43]: changes from 5540

File: src/main/modules/i18n/locales/cy.json [hunk 2]
```diff
@@ -2,27 +2,42 @@
   "COMMON": {
     "ACCOUNT_TYPE": "Math o gyfrif",
     "ADDRESS": "Cyfeiriad",
+    "ANNUAL_TURNOVER": "Trosiant blynyddol",
```
Comment (laperla11) [inline src/main/modules/i18n/locales/cy.json:5]: changes from 5540

File: src/main/modules/i18n/locales/en.json [hunk 3]
```diff
@@ -2,26 +2,41 @@
   "COMMON": {
     "ACCOUNT_TYPE": "Type of account",
     "ADDRESS": "Address",
+    "ANNUAL_TURNOVER": "Annual Turnover",
```
Comment (laperla11) [inline src/main/modules/i18n/locales/en.json:5]: changes from 5540

File: src/main/services/features/claimantResponse/claimantResponseService.ts [hunk 4]
```diff
@@ -1,5 +1,18 @@
 import {ClaimantResponse} from '../../../common/models/claimantResponse';
 import {getCaseDataFromStore, saveDraftClaim} from '../../../modules/draft-store/draftStoreService';
+import {Claim} from '../../../common/models/claim';
```
Comment (laperla11) [inline src/main/services/features/claimantResponse/claimantResponseService.ts:3]: changes from 5540

File: src/main/services/features/response/checkAnswers/financialSection/answers/addBankAccounts.ts [hunk 5]
```diff
@@ -18,7 +18,7 @@ export const addBankAccounts = (claim: Claim, financialSection: SummarySection,
   const yourBankAccountHref = CITIZEN_BANK_ACCOUNT_URL.replace(':id', claimId);
   if (claim.statementOfMeans?.bankAccounts) {
     const bankAccounts: CitizenBankAccount[] = claim.statementOfMeans.bankAccounts;
-    financialSection.summaryList.rows.push(summaryRow(t('PAGES.CHECK_YOUR_ANSWER.BANK_AND_SAVINGS_ACCOUNTS', {lng: getLng(lang)}), '', yourBankAccountHref, changeLabel(lang)));
+    financialSection.summaryList.rows.push(summaryRow(t('COMMON.BANK_AND_SAVINGS_ACCOUNTS', {lng: getLng(lang)}), '', yourBankAccountHref, changeLabel(lang)));
```
Comment (laperla11) [inline src/main/services/features/response/checkAnswers/financialSection/answers/addBankAccounts.ts:21]: changes from 5540

File: src/main/services/features/response/checkAnswers/financialSection/answers/addDependants.ts [hunk 6]
```diff
@@ -16,7 +16,7 @@ export const addDependants = (claim: Claim, financialSection: SummarySection, cl
   const numberOfChildren = claim.statementOfMeans?.dependants?.numberOfChildren;
   const numberOfChildrenLivingWithYou = claim.statementOfMeans?.numberOfChildrenLivingWithYou;
 
-  financialSection.summaryList.rows.push(summaryRow(t('PAGES.CHECK_YOUR_ANSWER.CHILDREN', { lng: getLng(lang) }), '', yourDependantsHref, changeLabel(lang)));
+  financialSection.summaryList.rows.push(summaryRow(t('COMMON.CHILDREN', { lng: getLng(lang) }), '', yourDependantsHref, changeLabel(lang)));
```
Comment (laperla11) [inline src/main/services/features/response/checkAnswers/financialSection/answers/addDependants.ts:19]: changes from 5540

File: src/main/services/features/response/checkAnswers/financialSection/answers/addEmploymentDetails.ts [hunk 7]
```diff
@@ -39,7 +39,7 @@ const showSelfEmploymentTaxPayments = (claim: Claim, financialSection: SummarySe
   if (taxPayments?.owed) {
     financialSection.summaryList.rows.push(summaryRow(t('PAGES.CHECK_YOUR_ANSWER.TAX_PAYMENT_ARE_YOU_BEHIND', { lng: getLng(lang) }), t(`COMMON.${isBehindTaxPayments}`, {lng: getLng(lang)}), '', changeLabel(lang)));
     financialSection.summaryList.rows.push(summaryRow(t('PAGES.CHECK_YOUR_ANSWER.TAX_PAYMENT_AMOUNT_YOU_OWE', { lng: getLng(lang) }), currencyFormatWithNoTrailingZeros(taxPayments.amountOwed), '', changeLabel(lang)));
-    financialSection.summaryList.rows.push(summaryRow(t('PAGES.CHECK_YOUR_ANSWER.TAX_PAYMENT_REASON', { lng: getLng(lang) }), taxPayments.reason, '', changeLabel(lang)));
+    financialSection.summaryList.rows.push(summaryRow(t('COMMON.REASON', { lng: getLng(lang) }), taxPayments.reason, '', changeLabel(lang)));
```
Comment (laperla11) [inline src/main/services/features/response/checkAnswers/financialSection/answers/addEmploymentDetails.ts:42]: changes from 5540

File: src/main/services/features/response/checkAnswers/financialSection/answers/addLoansOrCreditCardDebts.ts [hunk 8]
```diff
@@ -21,9 +21,9 @@ export const addLoansOrCreditCardDebts = (claim: Claim, financialSection: Summar
     financialSection.summaryList.rows.push(summaryRow(t('PAGES.CHECK_YOUR_ANSWER.DEBTS_LOANS_OR_CREDIT_CARDS', { lng: getLng(lang) }), t(`COMMON.${option}`, {lng: getLng(lang)}), yourLoansOrCreditCardsDebtsHref, changeLabel(lang)));
     for (let i = 0; i < debtsItems.length; i++) {
       financialSection.summaryList.rows.push(
-        summaryRow((debtsItems.length > 1 ? (i + 1) + '. ' : '') + t('PAGES.CHECK_YOUR_ANSWER.DEBT', { lng: getLng(lang) }), debtsItems[i].debt, '', changeLabel(lang)),
-        summaryRow(t('PAGES.CHECK_YOUR_ANSWER.DEBTS_TOTAL_OWED', { lng: getLng(lang) }), currencyFormatWithNoTrailingZeros(Number(debtsItems[i].totalOwned)), '', changeLabel(lang)),
-        summaryRow(t('PAGES.CHECK_YOUR_ANSWER.DEBTS_MONTHLY_PAYMENTS', { lng: getLng(lang) }), currencyFormatWithNoTrailingZeros(Number(debtsItems[i].monthlyPayments)), '', changeLabel(lang)),
+        summaryRow((debtsItems.length > 1 ? (i + 1) + '. ' : '') + t('COMMON.DEBT', { lng: getLng(lang) }), debtsItems[i].debt, '', changeLabel(lang)),
```
Comment (laperla11) [inline src/main/services/features/response/checkAnswers/financialSection/answers/addLoansOrCreditCardDebts.ts:24]: changes from 5540

File: src/main/services/features/response/checkAnswers/financialSection/answers/addOtherDependants.ts [hunk 9]
```diff
@@ -17,7 +17,7 @@ export const addOtherDependants = (claim: Claim, financialSection: SummarySectio
 
   if (otherDependantsOption === YesNo.YES) {
     financialSection.summaryList.rows.push(summaryRow(t('PAGES.CHECK_YOUR_ANSWER.SUPPORT_ANYONE_ELSE_FINANCIALLY', { lng: getLng(lang) }), '', yourOtherDependantsHref, changeLabel(lang)));
-    financialSection.summaryList.rows.push(summaryRow(t('PAGES.CHECK_YOUR_ANSWER.SUPPORT_NUMBER_OF_PEOPLE', { lng: getLng(lang) }), otherDependants.numberOfPeople.toString(), '', changeLabel(lang)));
-    financialSection.summaryList.rows.push(summaryRow(t('PAGES.CHECK_YOUR_ANSWER.SUPPORT_GIVE_DETAILS', { lng: getLng(lang) }), otherDependants.details, '', changeLabel(lang)));
+    financialSection.summaryList.rows.push(summaryRow(t('COMMON.NUMBER_OF_PEOPLE', { lng: getLng(lang) }), otherDependants.numberOfPeople.toString(), '', changeLabel(lang)));
```
Comment (laperla11) [inline src/main/services/features/response/checkAnswers/financialSection/answers/addOtherDependants.ts:20]: changes from 5540

File: src/main/services/genericService.ts [hunk 10]
```diff
@@ -0,0 +1,75 @@
+import {BankAccountTypeValues} from '../common/form/models/bankAndSavings/bankAccountTypeValues';
```
Comment (laperla11) [inline src/main/services/genericService.ts:1]: changes from 5540

File: src/main/views/features/response/statementOfMeans/citizenBankAndSavings/citizen-bank-accounts.njk [hunk 11]
```diff
@@ -32,10 +32,10 @@
             <label id="account-type-label" class="govuk-label--s">{{ t('COMMON.ACCOUNT_TYPE') }}</label>
           </div>
           <div class="govuk-grid-column-one-third govuk-!-display-inline">
-            <label id="account-joint-label" class="govuk-label--s">{{ t('PAGES.CITIZEN_BANK_ACCOUNTS.JOINT_ACCOUNT') }}</label>
+            <label id="account-joint-label" class="govuk-label--s">{{ t('COMMON.BANK_JOINT_ACCOUNT') }}</label>
```
Comment (laperla11) [inline src/main/views/features/response/statementOfMeans/citizenBankAndSavings/citizen-bank-accounts.njk:35]: changes from 5540

File: src/main/views/features/response/statementOfMeans/debts/debts.njk [hunk 12]
```diff
@@ -24,10 +24,10 @@
       <label id="debts-label" class="govuk-label--s">{{ t('PAGES.DEBTS.LIST') }}</label>
     </div>
     <div class="govuk-grid-column-one-quarter govuk-!-text-align-right">
-      <label id="total-owed-label" class="govuk-label--s ">{{ t('PAGES.DEBTS.TOTAL') }}</label>
+      <label id="total-owed-label" class="govuk-label--s ">{{ t('COMMON.TOTAL_DEBT') }}</label>
```
Comment (laperla11) [inline src/main/views/features/response/statementOfMeans/debts/debts.njk:27]: changes from 5540

File: src/test/unit/services/features/response/checkAnswers/financialDetails/addOtherDependants.test.ts [hunk 13]
```diff
@@ -36,10 +36,10 @@ describe('Other Dependants Details', () => {
     expect(summarySections.sections[constVal.INDEX_FINANCIAL_SECTION].summaryList.rows[6].actions?.items[0].href).toBe(CITIZEN_OTHER_DEPENDANTS_URL.replace(':id', constVal.CLAIM_ID));
     expect(summarySections.sections[constVal.INDEX_FINANCIAL_SECTION].summaryList.rows[6].actions?.items[0].text).toBe(constVal.PAGES_CHECK_YOUR_ANSWER_CHANGE);
 
-    expect(summarySections.sections[constVal.INDEX_FINANCIAL_SECTION].summaryList.rows[7].key.text).toBe('PAGES.CHECK_YOUR_ANSWER.SUPPORT_NUMBER_OF_PEOPLE');
+    expect(summarySections.sections[constVal.INDEX_FINANCIAL_SECTION].summaryList.rows[7].key.text).toBe('COMMON.NUMBER_OF_PEOPLE');
```
Comment (laperla11) [inline src/test/unit/services/features/response/checkAnswers/financialDetails/addOtherDependants.test.ts:39]: changes from 5540

File: src/main/common/form/models/claimSummarySection.ts [hunk 14]
```diff
@@ -1,4 +1,4 @@
-import {SummaryRow} from '../../../common/models/summaryList/summaryList';
+import {SummaryRow, TableCell} from '../../../common/models/summaryList/summaryList';
```
Comment (laperla11) [inline src/main/common/form/models/claimSummarySection.ts:1]: 5563 changes

File: src/main/common/models/summaryList/summaryList.ts [hunk 15]
```diff
@@ -52,3 +52,6 @@ export function summaryRow(key?: string, value?: string, href?: string, hrefText
   }
   return row;
 }
```
Comment (laperla11) [inline src/main/common/models/summaryList/summaryList.ts:54]: 5563 changes

File: src/main/common/form/models/claimSummarySection.ts [hunk 16]
```diff
@@ -1,4 +1,4 @@
...
```
Comment (laperla11) [inline src/main/common/form/models/claimSummarySection.ts:2]: 5563 changes

File: src/main/routes/features/claimantResponse/reviewDefendantsResponseController.ts [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+
+reviewDefendantsResponseController.get(CLAIMANT_RESPONSE_REVIEW_DEFENDANTS_RESPONSE_URL, async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const claimId = req.params.id;
+    const lang = req.query.lang ? req.query.lang : req.cookies.lang;
+    const claim: Claim = await getCaseDataFromStore(claimId);
+    const continueLink = constructResponseUrlWithIdParams(claimId, CLAIMANT_RESPONSE_TASK_LIST_URL);
+    // TODO: to be done after CIV-5793 is completed
+    const downloadResponseLink = '#';
+    const financialDetails = getFinancialDetails(claim, lang);
```
Comment (laperla11) [inline src/main/routes/features/claimantResponse/reviewDefendantsResponseController.ts:22]: 5563 changes

File: src/main/services/features/claimantResponse/defendantResponse/defendantResponseSummaryService.ts [hunk 18]
```diff
@@ -0,0 +1,14 @@
+import {Claim} from '../../../../common/models/claim';
```
Comment (laperla11) [inline src/main/services/features/claimantResponse/defendantResponse/defendantResponseSummaryService.ts:1]: 5563 changes

File: src/main/services/features/claimantResponse/defendantResponse/fullAdmissinionDefendantsResponseContent.ts [hunk 19]
```diff
@@ -0,0 +1,57 @@
+import {Claim} from '../../../../common/models/claim';
```
Comment (laperla11) [inline src/main/services/features/claimantResponse/defendantResponse/fullAdmissinionDefendantsResponseContent.ts:1]: 5563 changes

File: src/main/services/features/claimantResponse/defendantResponse/fullDisputeDefendantsResponseContent.ts [hunk 20]
```diff
@@ -0,0 +1,167 @@
+import {t} from 'i18next';
```
Comment (laperla11) [inline src/main/services/features/claimantResponse/defendantResponse/fullDisputeDefendantsResponseContent.ts:1]: 5563 changes

File: src/main/views/features/claimantResponse/review-defendants-response.njk [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+  <div class="govuk-width-container">
+    <main class="govuk-main-wrapper" role="main">
+      <div class="govuk-grid-row">
+        <div class="govuk-grid-column-two-thirds">
+          <h1 class="govuk-heading-l">{{ t('PAGES.REVIEW_DEFENDANTS_RESPONSE.TITLE') }}</h1>
+
+          {% for item in defendantsResponseContent %}
+            {{ itemContent(item,t) }}
+          {% endfor %}
+
```
Comment (laperla11) [inline src/main/views/features/claimantResponse/review-defendants-response.njk:22]: 5563 changes

File: src/main/views/features/dashboard/item-content.njk [hunk 22]
```diff
@@ -1,6 +1,7 @@
 {% from "govuk/components/button/macro.njk" import govukButton %}
```
Comment (laperla11) [inline src/main/views/features/dashboard/item-content.njk:1]: 5563 changes

File: src/test/unit/services/features/claimantResponse/defendantResponse/defendantResponseSummaryService.test.ts [hunk 23]
```diff
@@ -0,0 +1,85 @@
+import {Claim} from '../../../../../../main/common/models/claim';
```
Comment (laperla11) [inline src/test/unit/services/features/claimantResponse/defendantResponse/defendantResponseSummaryService.test.ts:1]: 5563 changes

**Review by neeta-hmcts — APPROVED** [review]

**Review by TatsyMaslova — APPROVED** [review]

**Review by portizdev — APPROVED** [review]

**Comment by laperla11** [comment]
created a new one because of pipeline issues;
https://github.com/hmcts/civil-citizen-ui/pull/1117

**Pull request closed without merging**